paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java

```diff
@@ -195,6 +195,11 @@ private void checkImmutability(Map<String, String> dynamicOptions) {
                 (k, v) -> {
                     if (!Objects.equals(v, options.get(k))) {
                         SchemaManager.checkAlterTableOption(k);
+
+                        if (CoreOptions.BUCKET.key().equalsIgnoreCase(k)) {
```

Comment:
should check equality between old and new

Reply:
Have checked:
```
if (!Objects.equals(v, options.get(k))) {
   xxx
}
```